src/main.rs

```diff
@@ -16,20 +17,32 @@ struct AppState {
     pages: Arc<HashMap<String, String>>,
 }
 
+fn parse_md_to_html(markdown_input: &str) -> String {
+    let mut options = Options::empty();
+    options.insert(Options::ENABLE_STRIKETHROUGH);
+    let parser = Parser::new_ext(markdown_input, options);
+
+    let mut html_output = String::new();
+    html::push_html(&mut html_output, parser);
+    html_output
+}
+
 fn read_content(path: &Path) -> Result<HashMap<String, String>, Error> {
     let mut pages = HashMap::new();
     match read_dir(path) {
         Ok(files) => {
             for file in files {
                 let file_path = file?.path();
+                let markdown_input = &*read_to_string(&file_path)?;
```

Comment:
This isn't wrong, but it kind of confused me. The `&*` pattern is something you do see when working with strings, but I don't think it's necessary here and it makes this more awkward to read.

Instead of the `&*` to dereference the `String` to a `str` and then borrow it for a `&str`, I would just store the `String` in the variable here... 

```suggestion
                let markdown_input = read_to_string(&file_path)?;
```

Reply:
this gives the following error:
`
   Compiling wirebrush v0.1.0 (/home/gem/source/wirebrush)
error[E0308]: mismatched types
  --> src/main.rs:45:38
   |
45 |                     parse_md_to_html(markdown_input),
   |                     ---------------- ^^^^^^^^^^^^^^
   |                     |                |
   |                     |                expected `&str`, found `String`
   |                     |                help: consider borrowing here: `&markdown_input`
   |                     arguments to this function are incorrect
   |
note: function defined here
  --> src/main.rs:20:4
   |
20 | fn parse_md_to_html(markdown_input: &str) -> String {
   |    ^^^^^^^^^^^^^^^^ --------------------

For more information about this error, try `rustc --explain E0308`.
error: could not compile `wirebrush` due to previous error
`